jsapp/js/projects/myOrgProjectsRoute.tsx

```diff
@@ -0,0 +1,256 @@
+// Libraries
+import React, {useState, useEffect} from 'react';
+import {observer} from 'mobx-react-lite';
+import {toJS} from 'mobx';
+import {useSearchParams, useNavigate} from 'react-router-dom';
+
+// Partial components
+import ProjectsFilter from './projectViews/projectsFilter';
+import ProjectsFieldsSelector from './projectViews/projectsFieldsSelector';
+import ViewSwitcher from './projectViews/viewSwitcher';
+import ProjectsTable from 'js/projects/projectsTable/projectsTable';
+import ProjectQuickActionsEmpty from './projectsTable/projectQuickActionsEmpty';
+import ProjectQuickActions from './projectsTable/projectQuickActions';
+import ProjectBulkActions from './projectsTable/projectBulkActions';
+import Icon from 'js/components/common/icon';
+import LimitNotifications from 'js/components/usageLimits/limitNotifications.component';
+import TransferProjectsInvite from 'js/components/permissions/transferProjects/transferProjectsInvite.component';
+import Button from 'js/components/common/button';
+import LoadingSpinner from 'js/components/common/loadingSpinner';
+
+// Stores, hooks and utilities
+import customViewStore from './customViewStore';
+import {
+  isInviteForLoggedInUser,
+  TransferStatuses,
+} from 'js/components/permissions/transferProjects/transferProjects.api';
+import {useOrganizationQuery} from 'js/account/stripe.api';
+
+// Constants and types
+import type {
+  ProjectsFilterDefinition,
+  ProjectFieldName,
+} from './projectViews/constants';
+import {
+  ORG_VIEW,
+  HOME_ORDERABLE_FIELDS,
+  HOME_DEFAULT_VISIBLE_FIELDS,
+  HOME_EXCLUDED_FIELDS,
+} from './projectViews/constants';
+import {ROOT_URL} from 'js/constants';
+import {endpoints} from 'js/api.endpoints';
+import {PROJECTS_ROUTES} from 'js/router/routerConstants';
+
+// Styles
+import styles from './projectViews.module.scss';
+
+interface InviteState {
```

Comment:
@jamesrkiger the whole invite related code is copy-pasted from `myProjectsRoute.tsx`. I wonder if I should componentize it and reuse in both, or should I rather remove id from `myOrgProjectsRoute.tsx`, and have the invite feature accessible in "My Projects" only.

Reply:
@magicznyleszek I think the invite code should be abstracted out. But on a broader note: Would it be possible to refactor more of these very similar views (`myProjectsRoute`, `myOrgProjectsRoute` and `customViewRoute`)?  At a glance, it looks like there's a lot of duplicated code we could reuse.

Reply:
I will see how possible it would be :)

Reply:
@jamesrkiger These are the unique pieces of each of the three components:

- dropzone: `MyProjectsRoute`
- invites: `MyProjectsRoute`, `MyOrgProjectsRoute`
- loading spinner: `MyOrgProjectsRoute`
- organization query: `MyOrgProjectsRoute`
- useEffect to navigate if org not MMO: `MyOrgProjectsRoute`
- export button: `CustomViewRoute`
- projectViewsStore: `CustomViewRoute`

Reply:
I think it would be possible to create some shared component, either for just `MyProjectsRoute` + `MyOrgProjectsRoute`, or maybe for all three. I would need to separate dropzone, invites, and export button first, so that each of these could be a component to be imported

Reply:
Sounds good, thank you. If you end up doing this, can you put the initial refactor in a separate PR to go along with our new PR guidelines?